### ISSUE TITLE
lane_manager.rb:106:in `choose_lane': [!] uninitialized constant Fastlane::LaneManager::Terminal (NameError)

### ISSUE THREAD

**Issue by gradha** [fastlane/fastlane#12293]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

After upgrading fastlane from 2.87-0 to 2.90.0 running any lane command fails. Running fastlane alone fails too. Removing lanes from the `Fastfile` makes the error go away, but is not a very useful workaround.

It would be nice to have a command to *downgrade* fastlane versions. **EDIT**: Ah, recovered 2.87 from time machine and now everything works again.

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>[0:gradha@sera.home:0] [~/Desktop/test]$ fastlane
[19:01:50]: ------------------------------
[19:01:50]: --- Step: default_platform ---
[19:01:50]: ------------------------------

Looking for related GitHub issues on fastlane/fastlane...

➡️  S3 Action is not able to find Access Key and Secret Access Key
    https://github.com/fastlane/fastlane/issues/4295 [closed] 22 💬
    12 Jan 2017

➡️  Remove Shenzhen dependency
    https://github.com/fastlane/fastlane/pull/6991 [closed] 19 💬
    13 Mar 2017

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/fastlane/lib/fastlane/lane_manager.rb:106:in `choose_lane': [!] uninitialized constant Fastlane::LaneManager::Terminal (NameError)
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/fastlane/lib/fastlane/lane_manager.rb:43:in `cruise_lane'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/fastlane/lib/fastlane/cli_tools_distributor.rb:107:in `take_off'
	from /Users/gradha/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/bin/fastlane:20:in `<top (required)>'
	from /Users/gradha/.fastlane/bin/bundle/bin/fastlane:22:in `load'
	from /Users/gradha/.fastlane/bin/bundle/bin/fastlane:22:in `<main>'
</pre>
</details>

### Environment

<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.4                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.9.1                           |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.13.4 (17E199)                   |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | true                                        |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.3                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  desc "Description of what the lane does"
  lane :custom_lane do
    # add actions here: https://docs.fastlane.tools/actions
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# app_identifier("[[APP_IDENTIFIER]]") # The bundle identifier of your app
# apple_id("[[APPLE_ID]]") # Your Apple email address


# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.90.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| CFPropertyList            | 3.0.0        |
| addressable               | 2.5.2        |
| atomos                    | 0.1.2        |
| babosa                    | 1.0.2        |
| bigdecimal                | 1.2.6        |
| bundler                   | 1.14.6       |
| claide                    | 1.0.2        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.2        |
| excon                     | 0.62.0       |
| faraday                   | 0.12.1       |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.1.3        |
| google-api-client         | 0.13.6       |
| googleauth                | 0.5.1        |
| highline                  | 1.7.10       |
| http-cookie               | 1.0.3        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| io-console                | 0.4.3        |
| json                      | 1.8.1        |
| jwt                       | 2.1.0        |
| jwt                       | 1.5.6        |
| libxml-ruby               | 3.0.0        |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| mime-types                | 3.1          |
| mime-types-data           | 3.2016.0521  |
| mini_magick               | 4.5.1        |
| minitest                  | 5.4.3        |
| multi_json                | 1.13.1       |
| multi_xml                 | 0.6.0        |
| multipart-post            | 2.0.0        |
| nanaimo                   | 0.2.5        |
| naturally                 | 2.1.0        |
| nokogiri                  | 1.5.6        |
| os                        | 0.9.6        |
| plist                     | 3.4.0        |
| power_assert              | 0.2.2        |
| psych                     | 2.0.8        |
| public_suffix             | 2.0.5        |
| rake                      | 10.4.2       |
| rdoc                      | 4.2.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| rouge                     | 2.0.7        |
| rouge                     | 1.11.1       |
| rubygems-update           | 2.5.2        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| signet                    | 0.7.3        |
| simctl                    | 1.6.3        |
| slack-notifier            | 2.3.2        |
| sqlite3                   | 1.3.13       |
| terminal-notifier         | 1.8.0        |
| terminal-table            | 1.7.3        |
| test-unit                 | 3.0.8        |
| tty-cursor                | 0.5.0        |
| tty-screen                | 0.6.4        |
| tty-spinner               | 0.8.0        |
| uber                      | 0.1.0        |
| unf                       | 0.1.4        |
| unf_ext                   | 0.0.7.4      |
| unicode-display_width     | 1.3.0        |
| unicode-display_width     | 1.1.3        |
| word_wrap                 | 1.0.0        |
| xcodeproj                 | 1.5.7        |
| xcpretty                  | 0.2.6        |
| xcpretty-travis-formatter | 0.0.4        |
</details>

*generated on:* **2018-04-12**
</details></pre>
</details>

**Comment by sushant40** [comment]
I have the same issue. I upgraded from 2.89.0

**Comment by sushant40** [comment]
hi @gradha ... you can downgrade by using following command... 

`sudo gem install fastlane -v 2.89.0`

then remove the bad version using... 

`sudo gem uninstall fastlane`

**Comment by revolter** [comment]
Or if you use `bundler`, you can specify the version like:

```ruby
gem 'fastlane', '2.89.0'
```

then run:

```bash
bundle update fastlane
```

After this, you can remove the version.

**Comment by raulrom13** [comment]
I have the same issue with version 2.90.0.

`Looking for related GitHub issues on fastlane/fastlane...

➡️  lane_manager.rb:106:in `choose_lane': [!] uninitialized constant Fastlane::LaneManager::Terminal (NameError)
    https://github.com/fastlane/fastlane/issues/12293 [open] 3 💬
    a minute ago

➡️  S3 Action is not able to find Access Key and Secret Access Key
    https://github.com/fastlane/fastlane/issues/4295 [closed] 22 💬
    12 Jan 2017

➡️  Remove Shenzhen dependency
    https://github.com/fastlane/fastlane/pull/6991 [closed] 19 💬
    13 Mar 2017

🔗  You can ⌘ + double-click on links to open them directly in your browser.
bundler: failed to load command: fastlane (/Users/raulrom13/.gems/bin/fastlane)
NameError: [!] uninitialized constant Fastlane::LaneManager::Terminal
  /Users/raulrom13/.gems/gems/fastlane-2.90.0/fastlane/lib/fastlane/lane_manager.rb:106:in `choose_lane'
  /Users/raulrom13/.gems/gems/fastlane-2.90.0/fastlane/lib/fastlane/lane_manager.rb:43:in `cruise_lane'
  /Users/raulrom13/.gems/gems/fastlane-2.90.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/raulrom13/.gems/gems/fastlane-2.90.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/raulrom13/.gem/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/raulrom13/.gem/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/raulrom13/.gem/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/raulrom13/.gems/gems/fastlane-2.90.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
  /Users/raulrom13/.gem/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/raulrom13/.gems/gems/fastlane-2.90.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
  /Users/raulrom13/.gems/gems/fastlane-2.90.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
  /Users/raulrom13/.gems/gems/fastlane-2.90.0/fastlane/lib/fastlane/cli_tools_distributor.rb:107:in `take_off'
  /Users/raulrom13/.gems/gems/fastlane-2.90.0/bin/fastlane:20:in `<top (required)>'
  /Users/raulrom13/.gems/bin/fastlane:22:in `load'
  /Users/raulrom13/.gems/bin/fastlane:22:in `<top (required)>'
________________________________________________________________________________
`

**Comment by revolter** [comment]
A fixed version will hopefully be released soon.

**Comment by zapjonny** [comment]
Me too!

**Comment by zapjonny** [comment]
If I ran `fastlane custom_lane` it worked. Just running `fastlane` crashes though.

> /.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.90.0/fastlane/lib/fastlane/lane_manager.rb:106:in `choose_lane': [!] uninitialized constant Fastlane::LaneManager::Terminal (NameError)

**Comment by dimmduh** [comment]
2.91 does not fix for me
"uninitialized constant Terminal"

**Comment by zapjonny** [comment]
2.91 fixed it for me.

**Comment by dimmduh** [comment]
@zapjonny I'm not rube programmer, may be something wrong in my code
29 line

`module Fastlane
  module Actions
    class UnityAction < Action
      def self.run(params)
	    executablePath = params[:executable]
		executablePath["{version}"] = params[:unity_version]
		
        build_cmd = "#{executablePath}"
        build_cmd << " -projectPath #{params[:project_path]}"
        build_cmd << " -batchmode"
        build_cmd << " -quit"
        build_cmd << " -logFile #{params[:log_path]}" if params[:log_path]

        build_cmd << " -nographics" if params[:nographics]

        build_cmd << " -executeMethod #{params[:execute_method]}" if params[:execute_method]

        build_cmd << " -runEditorTests" if params[:run_editor_tests]
        build_cmd << " -resultsFileDirectory=#{params[:results_file_directory]}" if params[:results_file_directory]
        build_cmd << " #{params[:additional_params]}"

        UI.message ""
        UI.message Terminal::Table123
		.new(
          title: "Unity".green,
          headings: ["Option", "Value"],
          rows: params.values
        )
        UI.message ""

        UI.message "Start running"
        UI.message "Check out logs if the build failed"
        UI.message ""

        sh build_cmd

        UI.success "Completed"
      end

      #####################################################
      # @!group Documentation
      #####################################################

      def self.description
        "Run Unity in batch mode"
      end

      def self.available_options
        [
          FastlaneCore::ConfigItem.new(key: :unity_version,
                                       env_name: "FL_UNITY_VERSION",
                                       description: "Version of unity",
                                       default_value: "2017.2.0f3"),
          FastlaneCore::ConfigItem.new(key: :executable,
                                       env_name: "FL_UNITY_EXECUTABLE",
                                       description: "Path for Unity executable",
                                       default_value: "/Applications/Unity_{version}/Unity.app/Contents/MacOS/Unity"),
          FastlaneCore::ConfigItem.new(key: :project_path,
                                       env_name: "FL_UNITY_PROJECT_PATH",
                                       description: "Path for Unity project",
                                       default_value: "#{Dir.pwd}"),

          FastlaneCore::ConfigItem.new(key: :execute_method,
                                       env_name: "FL_UNITY_EXECUTE_METHOD",
                                       description: "Method to execute",
                                       optional: true,
                                       default_value: nil),

          FastlaneCore::ConfigItem.new(key: :nographics,
                                       env_name: "FL_UNITY_NOGRAPHICS",
                                       description: "Initialize graphics device or not",
                                       is_string: false,
                                       default_value: true),

          FastlaneCore::ConfigItem.new(key: :run_editor_tests,
                                       env_name: "FL_UNITY_RUN_EDITOR_TESTS",
                                       description: "Option to run editor tests",
                                       is_string: false,
                                       default_value: false),
          FastlaneCore::ConfigItem.new(key: :results_file_directory,
                                       env_name: "FL_RESULTS_FILE_DIRECTORY",
                                       description: "Path for integration test results",
                                       optional: true,
                                       default_value: nil),
          FastlaneCore::ConfigItem.new(key: :log_path,
                                       env_name: "FL_LOG_FILE",
                                       description: "Path to log file",
                                       optional: true,
                                       default_value: nil),
          FastlaneCore::ConfigItem.new(key: :additional_params,
                                       description: "Additional params to unity",
                                       optional: true,
                                       default_value: "")
        ]
      end

      def self.authors
        ["dddnuts"]
      end

      def self.is_supported?(platform)
        [:ios].include?(platform)
      end
    end
  end
end
`

**Comment by revolter** [comment]
@dimmduh, Try adding `require 'terminal-table'` at the top of your file.

**Comment by tseglevskiy** [comment]
Folks, I have this problem for all versions from 2.90.0 till last one (2.97.0), 2.89.0 works fine. Seems like for most of you problem was fixed in 2.91.0. Is there any ideas, what to do? Is it problems with my environment?
